core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -301,6 +301,10 @@ final object Decoder extends TupleDecoders with LowPriorityDecoders {
     c.focus match {
       case JNull => Xor.right(Float.NaN)
       case JNumber(number) => Xor.right(number.toDouble.toFloat)
+      case JString(string) => Xor.fromOption(
+        JsonNumber.fromString(string).map(_.toDouble.toFloat),
```

Comment:
@knutwalker Thanks for pointing this out.

`Double` is kind of different from the integral decoders, though. `BiggerDecimal.toDouble` isn't partial, so it does truncate (or rather overflow to infinity) and can lose precision, and I think `Float` should probably follow suit here, although I'm happy to hear counter-arguments, of course. This behavior should be much better documented, too.